src/frontend/js/components/DashboardCreditCardsManagement/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import { useMemo } from 'react';
+import { DashboardCard } from 'components/DashboardCard';
+import { DashboardCreditCardBox } from 'components/DashboardCreditCardsManagement/DashboardCreditCardBox';
+import { CreditCard } from 'types/Joanie';
+import Banner, { BannerType } from 'components/Banner';
+import { Spinner } from 'components/Spinner';
+import { useCreditCardsManagement } from 'hooks/useCreditCardsManagement';
+
+const messages = defineMessages({
+  header: {
+    id: 'components.DashboardCreditCardsManagement.header',
+    description: 'Title of the dashboard credit cards management block',
+    defaultMessage: 'Credit cards',
+  },
+  errorCannotPromoteMain: {
+    id: 'components.DashboardCreditCardsManagement.errorCannotPromoteMain',
+    description: 'Error shown if a user tries to promote a main credit card',
+    defaultMessage: 'Cannot promote main credit card.',
+  },
+  emptyList: {
+    id: 'components.DashboardCreditCardsManagement.emptyList',
+    description: 'Empty placeholder of the dashboard credit cards management block',
+    defaultMessage: "You haven't created any credit cards yet.",
+  },
+});
+
+interface Props {
+  onClickEdit?: (creditCard: CreditCard) => void;
+}
+
+export const DashboardCreditCardsManagement = ({ onClickEdit }: Props) => {
+  const intl = useIntl();
+  const {
+    states: { error, isLoading },
+    methods: { setError, update, safeDelete },
+    ...creditCards
+  } = useCreditCardsManagement();
+
+  const sortByMainFirstThenByTitle = (a: CreditCard, b: CreditCard) => {
+    if (a.is_main) {
+      return -1;
+    }
+    if (b.is_main) {
+      return 1;
+    }
+    if (!a.title) {
+      return 1;
+    }
+    return a.title!.localeCompare(b.title ?? '', [intl.locale, intl.defaultLocale]);
+  };
```

Comment:
I think you should move this method outside of the component to prevent to recreate it at each render ? Just above the component is fine for me.

Reply:
If I move this function outside of the component then we will get an error due to the fact that this function uses `intl` 🙁